app/javascript/app/document-capture/components/form-steps.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+function FormSteps({ steps, onComplete }) {
+  const [values, setValues] = useState({});
+  const [stepIndex, setStepIndex] = useState(0);
+
+  function setStepValue(name, nextStepValue) {
+    setValues({ ...values, [name]: nextStepValue });
+  }
+
+  /**
+   * Increments state to the next step, or calls onComplete callback if the
+   * current step is the last step.
+   */
+  function toNextStep() {
+    const nextStepIndex = stepIndex + 1;
+    const isComplete = nextStepIndex === steps.length;
+    if (isComplete) {
+      // Clear step parameter from URL.
+      window.history.pushState(null, null, window.location.pathname);
+
+      onComplete(values);
+    } else {
+      const { name: nextStepName } = steps[nextStepIndex];
+
+      // Push the next step to history, both to update the URL, and to allow the
+      // user to return to an earlier step (see `popstate` sync behavior).
+      window.history.pushState(
+        { stepIndex: nextStepIndex },
+        nextStepName,
+        `?step=${nextStepName}`,
+      );
+
+      setStepIndex(nextStepIndex);
+    }
+  }
+
+  useEffect(() => {
+    function setStepIndexFromHistoryState(event) {
+      // Since there is no history state at the initial step, use 0 as default.
+      const { stepIndex: historyStepIndex = 0 } = event.state ?? {};
+      setStepIndex(historyStepIndex);
+    }
+
+    // If URL contains a step parameter on initial mount, it won't be possible
+    // to salvage the state, and the URL should be synced to the initial step
+    // state of the component (the first step).
+    if (window.location.search) {
+      window.history.replaceState(null, null, window.location.pathname);
+    }
+
+    window.addEventListener('popstate', setStepIndexFromHistoryState);
+    return () =>
+      window.removeEventListener('popstate', setStepIndexFromHistoryState);
+  }, []);
+
+  const step = steps[stepIndex];
+
+  // An empty steps array is allowed, in which case there is nothing to render.
+  if (!step) {
+    return null;
+  }
+
+  const { component: Component, name } = step;
+  const isLastStep = stepIndex + 1 === steps.length;
+
+  return (
+    <>
+      <Component
+        key={name}
+        value={values[name]}
+        onChange={(nextStepValue) => setStepValue(name, nextStepValue)}
+      />
+      <button type="button" onClick={toNextStep}>
+        {isLastStep ? 'Submit' : 'Continue'}
```

Comment:
Do we want to translate these? Or is that a later step with final UI?

Reply:
> Do we want to translate these? Or is that a later step with final UI?

🤷  It's not entirely clear to me, though it was on my mind as well (related to a [question I raised in Slack](https://gsa-tts.slack.com/archives/C0NGESUN5/p1595425398024400) and proposed handbook revisions).

I think in this case, since the strings are already readily available, and there's a pretty high likelihood they'd persist as the final copy, it'd probably be best to go ahead and translate them now.

https://github.com/18F/identity-idp/blob/8a263ed95c0d82408e7f4eb6108d53289fd6323f/config/locales/forms/en.yml#L51

https://github.com/18F/identity-idp/blob/8a263ed95c0d82408e7f4eb6108d53289fd6323f/config/locales/forms/en.yml#L41

Reply:
Yeah makes sense, especially for "standard" / stable copy like this, may as well translate now before we forget

Reply:
Updated in a39a01e.